packages/compartment-mapper/src/link.js

```diff
@@ -351,7 +353,10 @@ export const link = (
       parserForLanguage,
       moduleTransforms,
     );
-    const importHook = makeImportHook(location, name, parse);
+    /** @type {ShouldDeferError} */
+    const shouldDeferError = language =>
+      parserForLanguage[language || ''].heuristicImports; // TODO: make typescript happy without this empty string
```

Comment:
Haven't decided how to fix this yet. I don't think this is reachable with language===undefined, but makes sense to future proof it.